### ISSUE TITLE
[DOC-FIX] misleading deprecation of AI gateway in Python API mlflow.gateway documentation 

### ISSUE THREAD

**Issue by celeriev** [mlflow/mlflow#13876]
### Willingness to contribute

Yes. I would be willing to contribute a document fix with guidance from the MLflow community.

### URL(s) with the issue

The URL regarding the Python API MLFlow gateway, i.e. https://mlflow.org/docs/latest/python_api/mlflow.gateway.html is quite misleading. It is one of the first results when searching MLFlow gateway and suggests that it was deprecated in favor of MLFlow deployments, but MLFlow deployments was itself renamed AI Gateway.

Similarly, inside https://mlflow.org/docs/latest/cli.html#mlflow-deployments, under start-server, it is not clear that we should favor MLFlow gateway start

### Description of proposal (what needs changing)

Since the AI gateway starting with mlflow deployments start-server was deprecated in favor to mlflow gateway start we could improve the clarity of this choice in the documentation.

**Comment by WeichenXu123** [comment]
Yes we rename MLFlow deployments back to MLFlow gateway again. could you file a PR for updating it ?
